fn_cve_search/fn_cve_search/components/function_cve_search.py

```diff
@@ -0,0 +1,262 @@
+
+            # Getting Max Result Display Flag from app.config
+            MAX_RESULTS_RETURN = int(self.options.get('max_results_display'))
+
+            # Variables to Store Parsed CVE API Data
+            _result_search_data = dict()
+
+            log = logging.getLogger(__name__)
+            log.info("cve_search_data: %s", cve_search_data)
+            log.info("cve_search_criteria: %s", cve_search_criteria)
+            log.info("cve_id: %s", cve_id)
+            log.info("cve_vendor: %s", cve_vendor)
+            log.info("cve_product: %s", cve_product)
+            log.info("cve_published_date_from: %s", cve_published_date_from)
+            log.info("cve_published_date_to: %s", cve_published_date_to)
+
+            yield StatusMessage("starting...")
+
+            if cve_search_criteria.lower().find('search') != -1:
+                if cve_vendor is None and cve_product is None:
+                    raise FunctionError('Please Provide Vendor or Product Name')
+                else:
+                    _browse_data = self._search_cve_api(vendor_name=cve_vendor, product=cve_product)
+            elif cve_search_criteria.lower().find('specific') != -1:
+                _browse_data = self._get_specific_cve_data(cve_id=cve_id)
+            elif cve_search_criteria.lower().find('last') != -1:
+                _browse_data = self._last_30_cves()
+            else:
+                raise ValueError("CVE Browse Criteria is not recognized..!")
+            # Defining a key in result dictionary to store parsed data
+            _result_search_data['content'] = []
+
+            # Type Of the rest api call like browse,search,specific cve,last, db info
+            api_call_type = _browse_data['api_call']
+
+            # Rest Api Response Data
+            _browse_data_content = _browse_data['content']
+            if _browse_data_content:
+                if api_call_type == 'search':
+                    _result_search_data['api_call'] = 'search'
+                    search_data_list = self._parse_search_results(_browse_data_content, cve_published_date_from,
+                                                                  cve_published_date_to, MAX_RESULTS_RETURN)
+                    _result_search_data['content'].extend(search_data_list)
+                elif api_call_type == 'cve':
+                    _result_search_data['api_call'] = 'cve'
+                    cve_data_list = self._parse_cve_results(_browse_data_content)
+                    _result_search_data['content'].extend(cve_data_list)
+                elif api_call_type == 'last':
+                    _result_search_data['api_call'] = 'last'
+                    last_data_list = self._parse_last_cve_results(_browse_data_content, MAX_RESULTS_RETURN)
+                    _result_search_data['content'].extend(last_data_list)
+            log.debug("The Data Received from CVE DB : {}".format(_result_search_data))
+            yield StatusMessage("done...")
```

Comment:
"Successfully searched the CVE Database"

Reply:
ok